watertap/costing/units/electrodialysis.py

```diff
@@ -65,96 +47,72 @@ def cost_electrodialysis(blk, cost_electricity_flow=True):
     """
     t0 = blk.flowsheet().time.first()
 
-    membrane_cost = (
-        blk.costing_package.electrodialysis.cem_membrane_cost
-        + blk.costing_package.electrodialysis.aem_membrane_cost
-    )
-    spacer_cost = 2.0 * blk.costing_package.electrodialysis.flowspacer_cost
-    electrode_cost = 2.0 * blk.costing_package.electrodialysis.electrode_cost
-
-    cost_electrodialysis_stack(
-        blk,
-        membrane_cost,
-        spacer_cost,
-        blk.costing_package.electrodialysis.factor_membrane_housing_replacement,
-        electrode_cost,
-        blk.costing_package.electrodialysis.factor_electrode_replacement,
-    )
+    cost_electrodialysis_stack(blk)
 
     # Changed this to grab power from performance table which is identified
     # by same key regardless of whether the Electrodialysis unit is 0D or 1D
     if cost_electricity_flow:
-        blk.costing_package.cost_flow(
-            pyo.units.convert(
-                blk.unit_model.get_power_electrical(t0),
-                to_units=pyo.units.kW,
-            ),
-            "electricity",
-        )
+        if not has_rectifier:
+            blk.costing_package.cost_flow(
+                pyo.units.convert(
+                    blk.unit_model.get_power_electrical(t0),
+                    to_units=pyo.units.kW,
+                ),
+                "electricity",
+            )
+        else:
+            power = blk.unit_model.get_power_electrical(blk.flowsheet().time.first())
+            cost_rectifier(blk, power=power, ac_dc_conversion_efficiency=0.9)
```

Comment:
So with the current rectifier implementation, it is not possible to update the value of ac_dc_conversion_efficiency (its an expression). So besides passing has_rectifier, the user should be able to pass in the ac_dc_conversion_efficiency as well.

Reply:
I think the efficiency is in the argument too?